tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -231,6 +235,88 @@ def enroll_idm_and_configure_external_auth():
+
+
+def update_or_revert_rhsso_settings_in_satellite(action=None):
+    """Update the RH-SSO settings in satellite based on action"""
+    rhhso_settings = {
+        'authorize_login_delegation': True,
+        'authorize_login_delegation_auth_source_user_autocreate': 'External',
+        'oidc_algorithm': 'RS256',
+        'oidc_audience': '{}-foreman-openidc'.format(settings.server.hostname),
+        'oidc_issuer': '{}/auth/realms/{}'.format(settings.rhsso.host_url, settings.rhsso.realm),
+        'oidc_jwks_url': '{}/auth/realms/{}/protocol/openid-connect/certs'.format(
+            settings.rhsso.host_url, settings.rhsso.realm
+        )
+    }
+    if action == 'update':
+        for setting_name, setting_value in rhhso_settings.items():
+            setting_entity = entities.Setting().search(
+                query={'search': 'name={}'.format(setting_name)})[0]
+            setting_entity.value = setting_value
+            setting_entity.update({'value'})
+    else:
+        setting_entity = entities.Setting().search(
+            query={'search': 'name=authorize_login_delegation'})[0]
+        setting_entity.value = False
+        setting_entity.update({'value'})
+
+
+@fixture(scope='module')
+def enroll_configure_rhsso_external_auth():
+    """Enroll the Satellite6 Server to an RHSSO Server."""
+    run_command(cmd='yum -y --disableplugin=foreman-protector install '
+                    'mod_auth_openidc keycloak-httpd-client-install')
+    run_command(cmd='echo {0} | keycloak-httpd-client-install --app-name foreman-openidc \
+                --keycloak-server-url {1} \
+                --keycloak-admin-username "admin" \
+                --keycloak-realm "{2}" \
+                --keycloak-admin-realm master \
+                --keycloak-auth-role root-admin -t openidc -l /users/extlogin --force'.format(
+        settings.rhsso.password,
+        settings.rhsso.host_url,
+        settings.rhsso.realm,
+    ))
+    run_command(cmd="systemctl restart httpd")
+
+
+@fixture(scope='module')
+def enable_external_auth_rhsso(enroll_configure_rhsso_external_auth):
+    """register the satellite with RH-SSO Server for single sign-on"""
+    client_id = "{}{}".format(settings.server.hostname, '-foreman-openidc')
+    run_command(cmd="{0} config credentials "
+                    "--server {1}/auth "
+                    "--realm {2} "
+                    "--user {3} "
+                    "--password {4}".
+                format(KEY_CLOAK_CLI,
+                       settings.rhsso.host_url.replace("https://", "http://"),
+                       settings.rhsso.realm,
+                       settings.rhsso.rhsso_user,
+                       settings.rhsso.password),
+                hostname=settings.rhsso.host_name)
+
+    result = run_command(cmd=f"{KEY_CLOAK_CLI} get clients --fields id,clientId",
+                         hostname=settings.rhsso.host_name)
+    result_json = json.loads("[{{{0}".format("".join(result)))
```

Comment:
You mean to add the assertion before passing the JSON is correct?

Reply:
I mean in your string, `""[{{{0}"`, you open list (`[`) and dictionary as first element of list (`{`), but never close them. Which means that `}` and `]` must come from `result`, which in turn means that `result` itself is not valid JSON (since it lacks opening `[{`).

I don't know `results`, so I don't say it's wrong. Just feels odd.

Sorry, should have made that clear from beginning.

Reply:
Ohh Okay It's an issue with the Keyclaok command result, It gives the result JSON format, but somehow when I try to catch the result it is showed without having the ( [{ ) which is strange behaviour.

Reply:
Maybe there's some problem with that command, or our `run_command` wrapper. I wouldn't block PR on that, but we should get back to that and debug it, so we don't have to write something we don't like.

I guess jira card for automation functional team would be appropriate.